Wikipedia/Code/MWKImage.m

```diff
@@ -272,90 +218,34 @@ - (BOOL)isEqual:(id)object {
 }
 
 - (BOOL)isEqualToImage:(MWKImage*)image {
-    return self == image || [self.fileName isEqualToString:image.fileName];
+    return self == image || [[self.sourceURLString wmf_schemelessURL] isEqualToString:[image.sourceURLString wmf_schemelessURL]];
 }
 
 - (NSUInteger)hash {
     return [self.fileName hash];
 }
 
+- (BOOL)isPathEqualToImage:(MWKImage*)image {
+    return [[[self.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent] isEqualToString:[[image.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent]];
+}
+
 - (BOOL)isVariantOfImage:(MWKImage*)otherImage {
     // !!!: this might not be reliable due to underscore, percent encodings, and other unknowns w/ image filenames
     return [self.fileNameNoSizePrefix isEqualToString:otherImage.fileNameNoSizePrefix]
-           && ![self isEqualToImage:otherImage];
+           && [self isPathEqualToImage:otherImage];
```

Comment:
@joewalsh here is the "isVariantOfImage" test covering those "XXXpx-thumbnail.jpg" type image urls:
https://github.com/wikimedia/wikipedia-ios/pull/747/commits/c1da552df3dbdd851ab70ca8f25be8b4019144ef#diff-4f9ebe5e9be025960b5f61d5c676f7fbR36